val z: Boolean = true
val b: Byte = 0
val s: Short = 0
val i: Int = 0
val j: Long = 0L
val f: Float = 0.0f
val d: Double = 0.0
val c: Char = '0'

val nz: Boolean? = true
val nb: Byte? = 0
val ns: Short? = 0
val ni: Int? = 0
val nj: Long? = 0L
val nf: Float? = 0.0f
val nd: Double? = 0.0
val nc: Char? = '0'

val n: Number = 0
val nn: Number? = 0
val a: Any = 0
val na: Any? = 0

// Identity for primitive values of same type
val test_zz = <!FORBIDDEN_IDENTITY_EQUALS_WARNING!>z === z<!> || <!FORBIDDEN_IDENTITY_EQUALS_WARNING!>z !== z<!>
val test_bb = <!FORBIDDEN_IDENTITY_EQUALS_WARNING!>b === b<!> || <!FORBIDDEN_IDENTITY_EQUALS_WARNING!>b !== b<!>
val test_ss = <!FORBIDDEN_IDENTITY_EQUALS_WARNING!>s === s<!> || <!FORBIDDEN_IDENTITY_EQUALS_WARNING!>s !== s<!>
val test_ii = <!FORBIDDEN_IDENTITY_EQUALS_WARNING!>i === i<!> || <!FORBIDDEN_IDENTITY_EQUALS_WARNING!>i !== i<!>
val test_jj = <!FORBIDDEN_IDENTITY_EQUALS_WARNING!>j === j<!> || <!FORBIDDEN_IDENTITY_EQUALS_WARNING!>j !== j<!>
val test_ff = <!FORBIDDEN_IDENTITY_EQUALS_WARNING!>f === f<!> || <!FORBIDDEN_IDENTITY_EQUALS_WARNING!>f !== f<!>
val test_dd = <!FORBIDDEN_IDENTITY_EQUALS_WARNING!>d === d<!> || <!FORBIDDEN_IDENTITY_EQUALS_WARNING!>d !== d<!>
val test_cc = <!FORBIDDEN_IDENTITY_EQUALS_WARNING!>c === c<!> || <!FORBIDDEN_IDENTITY_EQUALS_WARNING!>c !== c<!>

// Identity for primitive values of different types (no extra error)
val test_zb = <!FORBIDDEN_IDENTITY_EQUALS!>z === b<!> || <!FORBIDDEN_IDENTITY_EQUALS!>z !== b<!>

// Primitive vs nullable
val test_znz = <!FORBIDDEN_IDENTITY_EQUALS_WARNING!>z === nz<!> || <!FORBIDDEN_IDENTITY_EQUALS_WARNING!>nz === z<!> || <!FORBIDDEN_IDENTITY_EQUALS_WARNING!>z !== nz<!> || <!FORBIDDEN_IDENTITY_EQUALS_WARNING!>nz !== z<!>
val test_bnb = <!FORBIDDEN_IDENTITY_EQUALS_WARNING!>b === nb<!> || <!FORBIDDEN_IDENTITY_EQUALS_WARNING!>nb === b<!> || <!FORBIDDEN_IDENTITY_EQUALS_WARNING!>b !== nb<!> || <!FORBIDDEN_IDENTITY_EQUALS_WARNING!>nb !== b<!>
val test_sns = <!FORBIDDEN_IDENTITY_EQUALS_WARNING!>s === ns<!> || <!FORBIDDEN_IDENTITY_EQUALS_WARNING!>ns === s<!> || <!FORBIDDEN_IDENTITY_EQUALS_WARNING!>s !== ns<!> || <!FORBIDDEN_IDENTITY_EQUALS_WARNING!>ns !== s<!>
val test_ini = <!FORBIDDEN_IDENTITY_EQUALS_WARNING!>i === ni<!> || <!FORBIDDEN_IDENTITY_EQUALS_WARNING!>ni === i<!> || <!FORBIDDEN_IDENTITY_EQUALS_WARNING!>i !== ni<!> || <!FORBIDDEN_IDENTITY_EQUALS_WARNING!>ni !== i<!>
val test_jnj = <!FORBIDDEN_IDENTITY_EQUALS_WARNING!>j === nj<!> || <!FORBIDDEN_IDENTITY_EQUALS_WARNING!>nj === j<!> || <!FORBIDDEN_IDENTITY_EQUALS_WARNING!>j !== nj<!> || <!FORBIDDEN_IDENTITY_EQUALS_WARNING!>nj !== j<!>
val test_fnf = <!FORBIDDEN_IDENTITY_EQUALS_WARNING!>f === nf<!> || <!FORBIDDEN_IDENTITY_EQUALS_WARNING!>nf === f<!> || <!FORBIDDEN_IDENTITY_EQUALS_WARNING!>f !== nf<!> || <!FORBIDDEN_IDENTITY_EQUALS_WARNING!>nf !== f<!>
val test_dnd = <!FORBIDDEN_IDENTITY_EQUALS_WARNING!>d === nd<!> || <!FORBIDDEN_IDENTITY_EQUALS_WARNING!>nd === d<!> || <!FORBIDDEN_IDENTITY_EQUALS_WARNING!>d !== nd<!> || <!FORBIDDEN_IDENTITY_EQUALS_WARNING!>nd !== d<!>
val test_cnc = <!FORBIDDEN_IDENTITY_EQUALS_WARNING!>c === nc<!> || <!FORBIDDEN_IDENTITY_EQUALS_WARNING!>nc === c<!> || <!FORBIDDEN_IDENTITY_EQUALS_WARNING!>c !== nc<!> || <!FORBIDDEN_IDENTITY_EQUALS_WARNING!>nc !== c<!>

// Primitive number vs Number
val test_bn = <!FORBIDDEN_IDENTITY_EQUALS_WARNING!>b === n<!> || <!FORBIDDEN_IDENTITY_EQUALS_WARNING!>n === b<!> || <!FORBIDDEN_IDENTITY_EQUALS_WARNING!>b !== n<!> || <!FORBIDDEN_IDENTITY_EQUALS_WARNING!>n !== b<!>
val test_sn = <!FORBIDDEN_IDENTITY_EQUALS_WARNING!>s === n<!> || <!FORBIDDEN_IDENTITY_EQUALS_WARNING!>n === s<!> || <!FORBIDDEN_IDENTITY_EQUALS_WARNING!>s !== n<!> || <!FORBIDDEN_IDENTITY_EQUALS_WARNING!>n !== s<!>
val test_in = <!FORBIDDEN_IDENTITY_EQUALS_WARNING!>i === n<!> || <!FORBIDDEN_IDENTITY_EQUALS_WARNING!>n === i<!> || <!FORBIDDEN_IDENTITY_EQUALS_WARNING!>i !== n<!> || <!FORBIDDEN_IDENTITY_EQUALS_WARNING!>n !== i<!>
val test_jn = <!FORBIDDEN_IDENTITY_EQUALS_WARNING!>j === n<!> || <!FORBIDDEN_IDENTITY_EQUALS_WARNING!>n === j<!> || <!FORBIDDEN_IDENTITY_EQUALS_WARNING!>j !== n<!> || <!FORBIDDEN_IDENTITY_EQUALS_WARNING!>n !== j<!>
val test_fn = <!FORBIDDEN_IDENTITY_EQUALS_WARNING!>f === n<!> || <!FORBIDDEN_IDENTITY_EQUALS_WARNING!>n === f<!> || <!FORBIDDEN_IDENTITY_EQUALS_WARNING!>f !== n<!> || <!FORBIDDEN_IDENTITY_EQUALS_WARNING!>n !== f<!>
val test_dn = <!FORBIDDEN_IDENTITY_EQUALS_WARNING!>d === n<!> || <!FORBIDDEN_IDENTITY_EQUALS_WARNING!>n === d<!> || <!FORBIDDEN_IDENTITY_EQUALS_WARNING!>d !== n<!> || <!FORBIDDEN_IDENTITY_EQUALS_WARNING!>n !== d<!>

// Primitive number vs Number?
val test_bnn = <!FORBIDDEN_IDENTITY_EQUALS_WARNING!>b === nn<!> || <!FORBIDDEN_IDENTITY_EQUALS_WARNING!>nn === b<!> || <!FORBIDDEN_IDENTITY_EQUALS_WARNING!>b !== nn<!> || <!FORBIDDEN_IDENTITY_EQUALS_WARNING!>nn !== b<!>
val test_snn = <!FORBIDDEN_IDENTITY_EQUALS_WARNING!>s === nn<!> || <!FORBIDDEN_IDENTITY_EQUALS_WARNING!>nn === s<!> || <!FORBIDDEN_IDENTITY_EQUALS_WARNING!>s !== nn<!> || <!FORBIDDEN_IDENTITY_EQUALS_WARNING!>nn !== s<!>
val test_inn = <!FORBIDDEN_IDENTITY_EQUALS_WARNING!>i === nn<!> || <!FORBIDDEN_IDENTITY_EQUALS_WARNING!>nn === i<!> || <!FORBIDDEN_IDENTITY_EQUALS_WARNING!>i !== nn<!> || <!FORBIDDEN_IDENTITY_EQUALS_WARNING!>nn !== i<!>
val test_jnn = <!FORBIDDEN_IDENTITY_EQUALS_WARNING!>j === nn<!> || <!FORBIDDEN_IDENTITY_EQUALS_WARNING!>nn === j<!> || <!FORBIDDEN_IDENTITY_EQUALS_WARNING!>j !== nn<!> || <!FORBIDDEN_IDENTITY_EQUALS_WARNING!>nn !== j<!>
val test_fnn = <!FORBIDDEN_IDENTITY_EQUALS_WARNING!>f === nn<!> || <!FORBIDDEN_IDENTITY_EQUALS_WARNING!>nn === f<!> || <!FORBIDDEN_IDENTITY_EQUALS_WARNING!>f !== nn<!> || <!FORBIDDEN_IDENTITY_EQUALS_WARNING!>nn !== f<!>
val test_dnn = <!FORBIDDEN_IDENTITY_EQUALS_WARNING!>d === nn<!> || <!FORBIDDEN_IDENTITY_EQUALS_WARNING!>nn === d<!> || <!FORBIDDEN_IDENTITY_EQUALS_WARNING!>d !== nn<!> || <!FORBIDDEN_IDENTITY_EQUALS_WARNING!>nn !== d<!>

// Primitive vs Any
val test_za = <!FORBIDDEN_IDENTITY_EQUALS_WARNING!>z === a<!> || <!FORBIDDEN_IDENTITY_EQUALS_WARNING!>a === z<!> || <!FORBIDDEN_IDENTITY_EQUALS_WARNING!>z !== a<!> || <!FORBIDDEN_IDENTITY_EQUALS_WARNING!>a !== z<!>
val test_ba = <!FORBIDDEN_IDENTITY_EQUALS_WARNING!>b === a<!> || <!FORBIDDEN_IDENTITY_EQUALS_WARNING!>a === b<!> || <!FORBIDDEN_IDENTITY_EQUALS_WARNING!>b !== a<!> || <!FORBIDDEN_IDENTITY_EQUALS_WARNING!>a !== b<!>
val test_sa = <!FORBIDDEN_IDENTITY_EQUALS_WARNING!>s === a<!> || <!FORBIDDEN_IDENTITY_EQUALS_WARNING!>a === s<!> || <!FORBIDDEN_IDENTITY_EQUALS_WARNING!>s !== a<!> || <!FORBIDDEN_IDENTITY_EQUALS_WARNING!>a !== s<!>
val test_ia = <!FORBIDDEN_IDENTITY_EQUALS_WARNING!>i === a<!> || <!FORBIDDEN_IDENTITY_EQUALS_WARNING!>a === i<!> || <!FORBIDDEN_IDENTITY_EQUALS_WARNING!>i !== a<!> || <!FORBIDDEN_IDENTITY_EQUALS_WARNING!>a !== i<!>
val test_ja = <!FORBIDDEN_IDENTITY_EQUALS_WARNING!>j === a<!> || <!FORBIDDEN_IDENTITY_EQUALS_WARNING!>a === j<!> || <!FORBIDDEN_IDENTITY_EQUALS_WARNING!>j !== a<!> || <!FORBIDDEN_IDENTITY_EQUALS_WARNING!>a !== j<!>
val test_fa = <!FORBIDDEN_IDENTITY_EQUALS_WARNING!>f === a<!> || <!FORBIDDEN_IDENTITY_EQUALS_WARNING!>a === f<!> || <!FORBIDDEN_IDENTITY_EQUALS_WARNING!>f !== a<!> || <!FORBIDDEN_IDENTITY_EQUALS_WARNING!>a !== f<!>
val test_da = <!FORBIDDEN_IDENTITY_EQUALS_WARNING!>d === a<!> || <!FORBIDDEN_IDENTITY_EQUALS_WARNING!>a === d<!> || <!FORBIDDEN_IDENTITY_EQUALS_WARNING!>d !== a<!> || <!FORBIDDEN_IDENTITY_EQUALS_WARNING!>a !== d<!>
val test_ca = <!FORBIDDEN_IDENTITY_EQUALS_WARNING!>c === a<!> || <!FORBIDDEN_IDENTITY_EQUALS_WARNING!>a === c<!> || <!FORBIDDEN_IDENTITY_EQUALS_WARNING!>c !== a<!> || <!FORBIDDEN_IDENTITY_EQUALS_WARNING!>a !== c<!>

// Primitive vs Any?
val test_zna = <!FORBIDDEN_IDENTITY_EQUALS_WARNING!>z === na<!> || <!FORBIDDEN_IDENTITY_EQUALS_WARNING!>na === z<!> || <!FORBIDDEN_IDENTITY_EQUALS_WARNING!>z !== na<!> || <!FORBIDDEN_IDENTITY_EQUALS_WARNING!>na !== z<!>
val test_bna = <!FORBIDDEN_IDENTITY_EQUALS_WARNING!>b === na<!> || <!FORBIDDEN_IDENTITY_EQUALS_WARNING!>na === b<!> || <!FORBIDDEN_IDENTITY_EQUALS_WARNING!>b !== na<!> || <!FORBIDDEN_IDENTITY_EQUALS_WARNING!>na !== b<!>
val test_sna = <!FORBIDDEN_IDENTITY_EQUALS_WARNING!>s === na<!> || <!FORBIDDEN_IDENTITY_EQUALS_WARNING!>na === s<!> || <!FORBIDDEN_IDENTITY_EQUALS_WARNING!>s !== na<!> || <!FORBIDDEN_IDENTITY_EQUALS_WARNING!>na !== s<!>
val test_ina = <!FORBIDDEN_IDENTITY_EQUALS_WARNING!>i === na<!> || <!FORBIDDEN_IDENTITY_EQUALS_WARNING!>na === i<!> || <!FORBIDDEN_IDENTITY_EQUALS_WARNING!>i !== na<!> || <!FORBIDDEN_IDENTITY_EQUALS_WARNING!>na !== i<!>
val test_jna = <!FORBIDDEN_IDENTITY_EQUALS_WARNING!>j === na<!> || <!FORBIDDEN_IDENTITY_EQUALS_WARNING!>na === j<!> || <!FORBIDDEN_IDENTITY_EQUALS_WARNING!>j !== na<!> || <!FORBIDDEN_IDENTITY_EQUALS_WARNING!>na !== j<!>
val test_fna = <!FORBIDDEN_IDENTITY_EQUALS_WARNING!>f === na<!> || <!FORBIDDEN_IDENTITY_EQUALS_WARNING!>na === f<!> || <!FORBIDDEN_IDENTITY_EQUALS_WARNING!>f !== na<!> || <!FORBIDDEN_IDENTITY_EQUALS_WARNING!>na !== f<!>
val test_dna = <!FORBIDDEN_IDENTITY_EQUALS_WARNING!>d === na<!> || <!FORBIDDEN_IDENTITY_EQUALS_WARNING!>na === d<!> || <!FORBIDDEN_IDENTITY_EQUALS_WARNING!>d !== na<!> || <!FORBIDDEN_IDENTITY_EQUALS_WARNING!>na !== d<!>
val test_cna = <!FORBIDDEN_IDENTITY_EQUALS_WARNING!>c === na<!> || <!FORBIDDEN_IDENTITY_EQUALS_WARNING!>na === c<!> || <!FORBIDDEN_IDENTITY_EQUALS_WARNING!>c !== na<!> || <!FORBIDDEN_IDENTITY_EQUALS_WARNING!>na !== c<!>
